### PR TITLE
Fixup block devs

### LINKED ISSUE #573: Use aggregator data instead of calling agent detect_scan.py and lustre.py
All the information required for detecting targets should now be contained within the output of the `device-aggregator` . The manager therefore should be able to derive the necessary information without calling the agent. This requires unwinding the plug-in process and re-factoring the calling code.

In the case of `detect_scan.py` re-factoring `DetectTargetsStep` to run locally on manager fetching and processing `device-aggregator` data.

### PR DESCRIPTION
**Pull request by jgrund**
Fixes #573 

Where possible use the information available in the `device-scanner` to derive required state and remove unnecessary code which calls out to the agent host shell. 

* Provide capability necessary through `block_devices.py` which can access `device-scanner` output.
* Add and update relevant tests 
* Implement relevant test fixes.

### CODE REVIEW

File: chroma-agent/chroma_agent/device_plugins/lustre.py [hunk 1]
```diff
@@ -107,6 +106,38 @@ def scan_packages():
     return repo_packages
 
 
+def process_zfs_mount(device, data, zfs_mounts):
+    # If zfs-backed target/dataset, lookup underlying pool to get uuid
+    # and nested dataset in zed structures to access lustre svname (label).
+    dev_root = device.split('/')[0]
```
Comment (johnsonw) [inline chroma-agent/chroma_agent/device_plugins/lustre.py]: If you have access to toolz you could do:
```
dev_root = toolz.pipe(
    device.split('/'),
    toolz.first
)
```

File: chroma-agent/tests/device_plugins/test_block_devices.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
+import json
+import os
+
+from django.utils import unittest
+from mock import patch
+
+from chroma_agent.device_plugins.block_devices import get_normalized_device_table, get_local_mounts
+
+
+# from chroma_core.services.plugin_runner import ResourceManager
```
Comment (johnsonw) [inline chroma-agent/tests/device_plugins/test_block_devices.py]: I guess the comments are temporary for now?

File: chroma-agent/chroma_agent/device_plugins/lustre.py [hunk 3]
```diff
@@ -121,32 +152,40 @@ def reset_state(self):
     def _scan_mounts(self):
         mounts = {}
 
-        for device, mntpnt, fstype in Mounts().all():
+        data = scanner_cmd("Stream")
+        local_mounts = get_local_mounts(data['localMounts'])
+        zfs_mounts = [(d, m, f) for d, m, f in local_mounts if f == 'zfs']
+
+        for device, mntpnt, fstype in local_mounts:
             if fstype != 'lustre':
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/lustre.py]: nit: Like you have `zfs_mounts` above you can do:

lustre_mounts = [x for x in local_mounts if f == 'lustre']

File: chroma-agent/chroma_agent/device_plugins/lustre.py [hunk 4]
```diff
@@ -121,32 +152,40 @@ def reset_state(self):
     def _scan_mounts(self):
         mounts = {}
 
-        for device, mntpnt, fstype in Mounts().all():
+        data = scanner_cmd("Stream")
+        local_mounts = get_local_mounts(data['localMounts'])
+        zfs_mounts = [(d, m, f) for d, m, f in local_mounts if f == 'zfs']
```
Comment (jgrund) [inline chroma-agent/chroma_agent/device_plugins/lustre.py]: Since you destructure and restructure here, isn't this just a noop?

**Pull request merged**